recipes-openxt/tss/tpm2-tools/tpm2-extendpcr-support.patch

```diff
@@ -1,19 +1,69 @@
-commit e4c796ddadda49305bb24249078accacca98446b
-Author: Chris Rogers <rogersc@ainfosec.com>
-Date:   Thu May 18 14:40:39 2017 -0400
+################################################################################
+SHORT DESCRIPTION:
+################################################################################
+Add tpm2_extendpcr binary.
 
-    [tpm2.0-tools] Extendpcr tool for OpenXT
-    
-      Implement new tool to extend a pcr of a specified bank with a
-      specified hash.
-    
-    Signed-off-by: Chris Rogers <rogersc@ainfosec.com>
+################################################################################
+LONG DESCRIPTION:
+################################################################################
 
-Index: tpm2-tools/tools/tpm2_extendpcr.c
-===================================================================
+################################################################################
+CHANGELOG
+################################################################################
+Add tpm2_extendpcr: Chris Rogers <rogersc@ainfosec.com>
+Uprev from v2.0.0 to v3.1.3: Nicholas Tsirakis <tsirakisn@ainfosec.com>
+
+################################################################################
+REMOVAL
+################################################################################
+The upstream tpm2-tools added their own tpm2_pcrextend tool in
+version 3.X.Y. This patch can be removed if that tool provides
+the same functionality, though it is currently untested.
```

Comment:
OpenXT should just use the upstream tpm2_pcrextend tool to align with upstream.  Any reason it wasn't tested?

Reply:
That's the eventual intention, but I believe their tpm2_pcrextend will still have to be patched a bit to fit our model for sealdata and unsealdata. I thought it was a bit too much for this PR, so I decided to save it for a future endeavor.

Reply:
In the end we need to get on a new version of the tools. With that said, I do agree with @jandryuk that the best solution would be to leverage as much of the upstream package as possible. Then any additional work we must do on top can be vetted with upstream project to see if we can get them to take our extensions.

Reply:
@dpsmith That's certainly the plan.  I'd prefer this step first to align versions while minimizing potential regressions.  Then the move to upstream pcrextend would be a small diff with a narrow focus for any regressions that may pop up.

Reply:
@crogers1, I guess from my perspective I would have done that as an incremental step during the development of the PR before publishing. If others are okay with seeing the churn in the main repo, then I have no objection.